src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -101,6 +110,28 @@ public EventBridgeWriter(EventBridgeSinkConfig config) {
     this.eventBridgeMapper = new DefaultEventBridgeMapper(config);
     this.batching = new DefaultEventBridgeBatching();
 
+    if ((config.offloadingS3defaultBucket != null) && !config.offloadingS3defaultBucket.isEmpty()) {
+      var s3client =
+          S3AsyncClient.builder()
+              .credentialsProvider(credentialsProvider)
+              .endpointOverride(endpointUri)
+              .forcePathStyle(endpointUri != null)
+              .httpClientBuilder(AwsCrtAsyncHttpClient.builder())
+              .overrideConfiguration(clientConfig)
+              .region(Region.of(this.config.region))
+              .build();
+      var bucketName = StringUtils.trim(config.offloadingS3defaultBucket);
+      var jsonPathExp = StringUtils.trim(config.offloadingDefaultFieldRef);
```

Comment:
see comment above: what if this is empty string now?

Reply:
While construct `S3EventBridgeEventDetailValueOffloading` there is a check for the field ref. The same check is used in `EventBridgeSinkConfigValidator` if the value is not null/empty.

Reply:
Well, IMHO this is what can happen:

- user specifies bucket but no fieldref
- offloading instance is created
- only once offloading is called during putevents we throw (in `transformToPutEventsRequestEntry` )

If so, IMHO this is quite late, no?

Reply:
The check is within the construction of the S3 offloading class.